lib/rules.js

```diff
@@ -184,6 +227,185 @@ export const MULTIPLY_BY_INVERSE = defineRuleString('#a / (#b / #c)', '#a * (#c
+        }
+        return (isDistributed && canEvaluate) ? {node} : null
+    },
+
+    (node) => {
+        const result = build.applyNode(
+            'mul',
+            node.args.map(nthRoot => {
+                let radicand = nthRoot.args[0]
+                let index = nthRoot.args[1]
+                console.log(index)
+                if (query.isNumber(radicand)) {
+                    return applyRule(NTH_ROOT_VALUE,
+                                     build.applyNode(
+                                         'nthRoot',
+                                         [radicand, index]
+                                     ))
+                } else {
+                    return applyRule(CANCEL_EXPONENT,
+                                     build.applyNode(
+                                         'nthRoot',
+                                         [radicand, index]
+                                     ))
+                }
+            })
+        )
+        return result
+    }
+)
+
+// e.g. 12 -> 2 * 2 * 3
+export const FACTOR_INTO_PRIME = defineRule(
+    (node) => {
+        return query.isNumber(node) ? {node} : null
+    },
+
+    (node) => {
+        const factors = primeFactorization(query.getValue(node))
+        const result = build.applyNode(
+            'mul',
+            factors.map(build.numberNode)
+        )
+        return result
+    }
+)
+
+// e.g. nthRoot(2 * 2 * 2, 2) -> nthRoot((2 * 2) * 2)
+export const GROUP_TERMS_BY_ROOT = defineRule(
+    (node) => {
+        let canGroupTerms = false
+        canGroupTerms = canApplyRule(DISTRIBUTE_NTH_ROOT, node)
+        return canGroupTerms ? {node} : null
+    },
+
+    (node) => {
+        const radicand = node.args[0]
+        const index = node.args[1]
+
+        // dictionary storing the number of times a constant appears
+        // e.g. 2 * 2 * 2 => {'2': 3} , 2 appears 3 times
+        const count = {}
+        radicand.args.forEach(arg => {
+            let key = JSON.stringify(arg)
+            count[key] ? count[key]++ : count[key] = 1
+        })
+
+        const flatten = arr => arr.reduce(
+            (acc, val) => acc.concat(
+                Array.isArray(val) ? flatten(val) : val
+            ),
+            []
+        )
+
+        const result = build.applyNode(
+            'nthRoot',
+            [build.applyNode(
+                'mul',
+                flatten(Object.keys(count).map(key => {
+                    let leftover = count[key]
+                    let term = JSON.parse(key)
+                    const times = query.getValue(index)
+
+                    const args = []
+
+                    while (leftover - times > 0) {
+                        leftover -= times
+                        args.push(Array(times).fill(term))
+                    }
+                    leftover == 1
+                        ? args.push(term)
+                        : args.push(Array(leftover).fill(term))
```

Comment:
You might want to reformulate this as:
```
const arg = leftover === 1
   ? term
   : build.applyNode('mul', Array(leftover).fill(term))
args.push(arg)
```
and then skip the `args.forEach` below.